cmd/earthly/build_cmd.go

```diff
@@ -296,8 +297,30 @@ func (app *earthlyApp) actionBuildImp(cliCtx *cli.Context, flagArgs, nonFlagArgs
 			return errors.Wrapf(err, "read %s", app.envFile)
 		}
 	}
+	validEnvNames := cliutil.GetValidEnvNames(app.cliApp)
+	for k := range dotEnvMap {
+		if _, found := validEnvNames[k]; !found {
+			app.console.Warnf("unexpected env \"%s\": as of v0.7.0, --build-arg values must be defined in .arg (and --secret values in .secret)", k)
+		}
+	}
+
+	argMap, err := godotenv.Read(app.argFile)
+	if err != nil {
+		// ignore ErrNotExist when using default .env file
+		if app.argFile != defaultArgFile || !errors.Is(err, os.ErrNotExist) {
```

Comment:
The `cli.StringFlag` type has an `IsSet()` method. I'm not familiar enough with `urfave` to know for sure, but would that allow us to check `if app.argFile.IsSet()` instead of `if app.argFile != defaultArgFile`? It would be nice to still error if a user explicitly passes the default.

Reply:
good point -- I thought there was a reason for this, but I think it's only for the initial loading of `.env` which must occur before `urfave` is initialised; however, there's no reason I can think of for this code not to use `IsSet()`. I've made the switch.